synapsis/src/test/java/com/soen/synapsis/unit/appuser/connection/ConnectionServiceTest.java

```diff
@@ -0,0 +1,72 @@
+package com.soen.synapsis.unit.appuser.connection;
+
+import com.soen.synapsis.appuser.AppUser;
+import com.soen.synapsis.appuser.Role;
+import com.soen.synapsis.appuser.connection.ConnectionKey;
+import com.soen.synapsis.appuser.connection.ConnectionRepository;
+import com.soen.synapsis.appuser.connection.ConnectionService;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.platform.engine.support.hierarchical.ThrowableCollector;
+import org.mockito.Mock;
+import org.mockito.MockitoAnnotations;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+import static org.mockito.BDDMockito.given;
+import static org.mockito.BDDMockito.willThrow;
+import static org.mockito.Mockito.when;
+
+public class ConnectionServiceTest {
+
+    @Mock
+    private ConnectionService connectionService;
+
+    @Mock
+    private ConnectionRepository connectionRepository;
+
+    private AutoCloseable autoCloseable;
+
+    @BeforeEach
+    void setUp() {
+        autoCloseable = MockitoAnnotations.openMocks(this);
+        connectionService = new ConnectionService(connectionRepository);
+    }
+
+    @AfterEach
+    void tearDown() throws Exception {
+        autoCloseable.close();
+    }
+
+    @Test
+    void makeConnectionSuccessful() {
```

Comment:
We should ensure that the connection is added to the database here

Reply:
I don't think I understand.  Do you want me to call "connectionRepository.save(connection)" in this test method?

Reply:
You can see an example in the register user tests